networkmanager/src/main/java/com/dezen/riccardo/networkmanager/StringResource.java

```diff
@@ -64,4 +66,19 @@ public Bundle getExtras() {
     public static StringResource getDefaultInvalid(){
         return new StringResource("","");
     }
+
+    /**
+     * Method to parse a String Resource from a String
+     * @param string the String to parse
+     * @param separator the separator which divides key and value, must not be a character of the
+     *                  key nor of the value.
+     * @return The StringResource
+     * @throws IllegalArgumentException if the separator splits the String in more than two parts
+     */
+    public static StringResource parseString(String string, String separator) throws IllegalArgumentException{
+        String[] parts = string.split(separator);
+        if(parts.length > 2)
```

Comment:
Wouldn't it be better to have `parts.length != 2` ? Because if the separator isn't in the string, then, if I'm not mistaken, split returns an array containing one string that is the original one (therefore length will be = 1).

Reply:
good point